core/src/main/java/io/grpc/NameResolverRegistry.java

```diff
@@ -83,12 +83,12 @@ public synchronized void deregister(NameResolverProvider provider) {
   private synchronized void refreshProviders() {
     List<NameResolverProvider> providers = new ArrayList<NameResolverProvider>(allProviders);
     // sort() must be stable, as we prefer first-registered providers
-    Collections.sort(providers, new Comparator<NameResolverProvider>() {
+    Collections.sort(providers, Collections.reverseOrder(new Comparator<NameResolverProvider>() {
```

Comment:
Why not just reverse the expression in the body?

Reply:
The comparator is the natural ordering. It can be subtle to notice a - b vs b - a. Using reverseOrder makes it more explicit.

Reply:
Maybe add a comment here that sums it up.
````java
// Higher values/priorities first
````
IMO thats way more expressive than a reverse order somewhere in between.

Reply:
Added a comment (mirroring what is in ServiceProviders). My concern here is that the comment could be wrong... In fact, I considered a comment before and it _would_ have been wrong because I lost the reverseOrder during refactoring.